lightningd/options.c

```diff
@@ -610,16 +610,27 @@ void setup_color_and_alias(struct lightningd *ld)
 
 	if (!ld->alias) {
 		u64 adjective, noun;
+		char *name;
+		int taillen;
 
 		memcpy(&adjective, der+3, sizeof(adjective));
 		memcpy(&noun, der+3+sizeof(adjective), sizeof(noun));
 		noun %= ARRAY_SIZE(codename_noun);
 		adjective %= ARRAY_SIZE(codename_adjective);
+
+		/* Only use 32 characters */
+		name = tal_fmt(ld, "%s%s-",
+			       codename_adjective[adjective],
+			       codename_noun[noun]);
+		assert(strlen(name) < 32);
+		taillen = 32 - strlen(name);
+		if (taillen > strlen(version()))
+			taillen = strlen(version());
+		tal_append_fmt(&name, version() + strlen(version()) - taillen);
```

Comment:
Missing `, "%s"`? Granted the version will not contain formatting characters etc. but the principle...